aws/ecr/kops_ecr_user.tf

```diff
@@ -9,6 +16,49 @@ module "kops_ecr_user" {
   }
 }
 
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "default" {
+  statement {
+    effect  = "Allow"
+    actions = ["ecr:GetAuthorizationToken"]
+
+    resources = ["*"]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ecr:InitiateLayerUpload",
+      "ecr:UploadLayerPart",
+      "ecr:CompleteLayerUpload",
+      "ecr:PutImage",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:GetRepositoryPolicy",
+      "ecr:DescribeRepositories",
+      "ecr:ListImages",
+      "ecr:DescribeImages",
+      "ecr:BatchGetImage",
+    ]
+
+    resources = ["arn:aws:ecr::${data.aws_caller_identity.current.account_id}:repository/*"]
+  }
+}
+
+resource "aws_iam_policy" "default" {
+  name        = "${module.label.id}"
+  description = "Allow IAM Users to access ECR registries"
+  policy      = "${data.aws_iam_policy_document.default.json}"
+}
+
+resource "aws_iam_policy_attachment" "default" {
```

Comment:
better not to use `aws_iam_policy_attachment` at all, use `aws_iam_user_policy_attachment`.
see the WARNING here https://www.terraform.io/docs/providers/aws/r/iam_policy_attachment.html